src/backend/api/meals.js

```diff
@@ -1,15 +1,91 @@
 const express = require("express");
 const router = express.Router();
 const knex = require("../database");
-
-router.get("/", async (request, response) => {
+const requiredColumns = [
+  "title",
+  "description",
+  "location",
+  "when",
+  "max_reservations",
+  "created_date",
+  "price",
+];
+const valid = (input, array) => {
+  return array.every((c) => input.hasOwnProperty(c) && input[c] !== "");
+};
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
-  } catch (error) {
-    throw error;
+    const meals = await knex("meal").select("title");
+    res.status(200).json(meals);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+router.post("/", async (req, res) => {
+  const addMeal = req.body;
+  if (!valid(addMeal, requiredColumns)) {
+    return res.status(404).json("Missing required columns in the request body");
+  } else {
+    try {
+      await knex("meal").insert(addMeal);
+      res.status(201).json(`meal ${addMeal.title} added successfully`);
+    } catch (err) {
+      res.status(500).json(err);
+    }
+  }
+});
+router.get("/:id", async (req, res) => {
+  const id = req.params.id;
+  if (isNaN(id)) {
```

Comment:
also add a check for if 'id' exists and also add it to the rest of endpoints which receives `id` from `req.params`

Reply:
how do i make this check ? Error: ENOENT: no such file or directory, stat 'C:\Users\ba3ba3\Desktop\meal-sharing\dist\index.html' 
I keep having this error , also tried req.body.myParam